core/src/main/java/org/elasticsearch/index/engine/InternalEngine.java

```diff
@@ -505,7 +525,29 @@ public void refresh(String source) throws EngineException {
             throw new RefreshFailedEngineException(shardId, t);
         }
 
-        // TODO: maybe we should just put a scheduled job in threadPool?
+        mergeScheduler.refreshConfig();
+
+        // we must also refresh our internal searcher, so that a subsequent real-time get (which uses internal searcher) comes from the
```

Comment:
There were RT get tests that were angry, because apparently an "RT get from xlog" can sometimes fail (if there were mapping changes maybe, not sure?) where "RT get from the index" would succeed.  So if an explicit refresh is invoked I need to refresh both internal and external readers.